operators/src/source/gdal_source/mod.rs

```diff
@@ -442,31 +558,62 @@ where
             &query
         );
 
-        let meta_data = self.meta_data.loading_info(query).await?;
-
         debug!(
             "GdalSourceProcessor<{:?}> meta data loaded, took {:?}.",
             P::TYPE,
             start.elapsed()
         );
 
-        // TODO: evaluate if there are GeoTransforms with positive y-axis
-        // The "Pixel-space" starts at the top-left corner of a `GeoTransform`.
-        // Therefore, the pixel size on the x-axis is always increasing
         let spatial_resolution = query.spatial_resolution;
 
+        // A `GeoTransform` maps pixel space to world space.
+        // Usually a SRS has axis directions pointing "up" (y-axis) and "up" (y-axis).
+        // We are not aware of spatial reference systems where the x-axis points to the right.
+        // However, there are spatial reference systems where the y-axis points downwards.
+        // The standard "pixel-space" starts at the top-left corner of a `GeoTransform` and points down-right.
+        // Therefore, the pixel size on the x-axis is always increasing
         let pixel_size_x = spatial_resolution.x;
         debug_assert!(pixel_size_x.is_sign_positive());
-        // and the pixel size on  the y-axis is always decreasing
+        // and the y-axis should only be positive if the y-axis of the spatial reference system also "points down".
+        // NOTE: at the moment we do not allow "down pointing" y-axis.
         let pixel_size_y = spatial_resolution.y * -1.0;
         debug_assert!(pixel_size_y.is_sign_negative());
 
         let tiling_strategy = self
             .tiling_specification
             .strategy(pixel_size_x, pixel_size_y);
 
-        // TODO: what to do if loading info is empty?
-        let source_stream = stream::iter(meta_data.info);
+        let result_descriptor = self.meta_data.result_descriptor().await?;
+
+        let mut empty = false;
+
+        if let Some(data_spatial_bounds) = result_descriptor.bbox {
+            if !data_spatial_bounds.intersects(&query.spatial_bounds) {
+                debug!("query does not intersect spatial data bounds");
+                empty = true;
+            }
+        }
+
+        // TODO: use the time bounds to early return.
+        /*
```

Comment:
why is this commented out?

Reply:
it will generate too large empty time-intervals.

Reply:
it simply needs more work to enable the check